### PR TITLE
Initial refactoring work

### PR DESCRIPTION
**Pull request by pierluca**
Please browse by commit:
- 92ad4b6 is purely about identifying issues in the current code base
- 5b56aa1 adds TypeScript support (still untested) and should support the `import 'Models/Object/hash';` syntax
- f6bb281 contains the classes themselves
- 44cd5ef can be ignored, as it merely moves the folder to facilitate a later merge.

### CODE REVIEW

File: PopApp/websockets/LocalMockServer.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ const webSocketsServerPort = 8000;
 const webSocketServer = require('websocket').server;
 const http = require('http');
 
+// FIXME: Could this be re-used further ? If not, remove it entirely.
+
```
Comment (RaulinN) [inline PopApp/websockets/LocalMockServer.js:7]: Probably only useful for low-level websocket testing. It was also supposed to be used in unit tests, but GitHub actions have problems running the server

File: PopApp/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
+    /* Basic Options */
+    // "incremental": true,                   /* Enable incremental compilation */
+    "target": "esnext",                       /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */
+    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */
+    "lib": ["es6"],                           /* Specify library files to be included in the compilation. */
```
Comment (RaulinN) [inline PopApp/tsconfig.json]: Using `btoa` and `atob` result in a compilation error. Adding the DOM library to the tsconfig file like in [this post](https://stackoverflow.com/questions/42899209/ts2304cannot-find-name-btoa) solves the issue.

(I cannot "propose changes" directly in the PR. Could it be a permissions issue?)

File: PopApp/Model/Network/Method/catchup.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import { JsonRpcParams } from './jsonRpcParams';
+
+export class Subscribe extends JsonRpcParams {}
```
Comment (RaulinN) [inline PopApp/Model/Network/Method/catchup.ts]: "Subscribe" should read "Catchup"

File: PopApp/Model/Objects/base64.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        super(value);
+    }
+
+    public static encode(text: string) : Base64Data {
+        return new Base64Data(atob(text));
+    }
+
+    public decode(): string {
+        return btoa(this.valueOf());
+    }
```
Comment (RaulinN) [inline PopApp/Model/Objects/base64.ts]: `atob` and `btoa` are inversed

**Review by RaulinN — CHANGES_REQUESTED** [review]

File: PopApp/websockets/LocalMockServer.js (hunk 1 above)
Comment (pierluca) [reply PopApp/websockets/LocalMockServer.js:7]: With Gaurav we were envisioning the possibility of having something like this, both to test the backends and the frontends.

We'll have to think a bit more about it though.

Are you sure you didn't have a problem like this?
https://github.community/t/test-http-server-in-github-actions/17608/6

File: PopApp/tsconfig.json (hunk 2 above)
Comment (pierluca) [reply PopApp/tsconfig.json]: Done. 
I think the proposed changes being blocked is only due to the fact that the commit is "outdated", it's not a permission issue.

File: PopApp/Model/Network/Method/catchup.ts (hunk 3 above)
Comment (pierluca) [reply PopApp/Model/Network/Method/catchup.ts]: Oops. 👍

File: PopApp/Model/Objects/base64.ts (hunk 4 above)
Comment (pierluca) [reply PopApp/Model/Objects/base64.ts]: Oops (2). 👍

**Pull request closed without merging**